SharedPackages/DataBrokerProtectionCore/Sources/DataBrokerProtectionCore/CCF/DataBrokerProtectionFeature.swift

```diff
@@ -132,5 +141,25 @@ public struct DataBrokerProtectionFeature: Subfeature {
         Logger.action.log("Pushing into WebView: \(method.rawValue) params \(String(describing: params))")
 
         broker.push(method: method.rawValue, params: params, for: self, into: webView)
+
+        removeTimer()
+        timer = Timer.scheduledTimer(withTimeInterval: actionResponseTimeout, repeats: false) { [weak self] _ in
+            self?.handleTimeout(for: (params as? Params)?.state.action)
+        }
+    }
+
+    private func handleTimeout(for action: Action?) {
+        Logger.action.log("Action timeout: \(String(describing: action))")
+
+        removeTimer()
+        Task {
+            await delegate?.onError(error: DataBrokerProtectionError.actionFailed(actionID: action?.id ?? "unknown",
+                                                                                  message: "Request timed out"))
+        }
+    }
```

Comment:
TODO: Limit this to ExpectationAction for scan operations only to limit the impact.


I’ll follow up with newer commit to address this.

Reply:
Done